packages/cli-lib/github.js

```diff
@@ -117,7 +124,83 @@ async function cloneGitHubRepo(dest, type, repoName, sourceDir, options = {}) {
   return success;
 }
 
+async function getGitHubRepoContentsAtPath(repoName, path) {
+  const contentsRequestUrl = `https://api.github.com/repos/HubSpot/${repoName}/contents/${path}`;
+
+  return request.get(contentsRequestUrl, {
+    json: true,
+    headers: { ...DEFAULT_USER_AGENT_HEADERS, ...GITHUB_AUTH_HEADERS },
+  });
+}
+
+async function fetchGitHubRepoContentFromDownloadUrl(dest, downloadUrl) {
+  const resp = await request.get(downloadUrl, {
+    headers: { ...DEFAULT_USER_AGENT_HEADERS, ...GITHUB_AUTH_HEADERS },
+  });
+
+  fs.writeFileSync(dest, resp, 'utf8');
+}
+
+/**
+ * Writes files from a HubSpot public repository to the destination folder
+ * @param {String} dest - Dir to write contents to
+ * @param {String} repoName - Name of GitHub repository to fetch contents from
+ * @param {String} path - Path to obtain contents from within repository
+ * @returns {Boolean} `true` if successful, `false` otherwise.
+ */
+async function downloadGitHubRepoContents(
```

Comment:
Wondering for right-click create commands & any executing these programmatically; are these being redownloaded every call? Could we cache them client side to improve performance, since they'll likely not be changed often?

Reply:
I think we could certainly do that, but to me, I think that the download is small enough that it makes sense to leave it like this, so any updates to the remote contents will come through immediately, without any need to invalidate the cache.